src/main/java/com/mongodb/kafka/connect/source/MongoSourceTask.java

```diff
@@ -171,17 +226,18 @@ public synchronized void stop() {
     MongoCursor<BsonDocument> createCursor(final MongoSourceConfig cfg, final MongoClient mongoClient) {
         LOGGER.debug("Creating a MongoCursor");
         ChangeStreamIterable<Document> changeStream = getChangeStreamIterable(cfg, mongoClient);
-
-        Map<String, Object> offset = context != null ? context.offsetStorageReader().offset(createPartitionMap(cfg)) : null;
-        if (offset != null) {
+        BsonDocument resumeAfter = getResumeAfter(cfg);
+        if (resumeAfter != null) {
             LOGGER.info("Resuming the change stream at the previous offset");
-            changeStream.resumeAfter(BsonDocument.parse((String) offset.get("_id")));
+            changeStream.resumeAfter(resumeAfter);
```

Comment:
Are there any tests of the overlap here?  I.E, a test that adds data while the copy is happening, that would produce duplicate inserts?

Reply:
No, it would be a racy test as we'd have to time the copying and insertion of data while the copying occurs.

Reply:
True, it would be racy.  OK, I'll resolve.